decidim-core/lib/tasks/procfile.rake

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require "thor"
+
+namespace :decidim do
+  namespace :procfile do
+    desc "Generates a dummy app for testing in external installations"
+    task :install do
+      actions :create_file, "Procfile.dev", <<~RUBY
+        web: bin/rails server -b 0.0.0.0 -p 3000
+        shakapacker: bin/webpacker-dev-server
+      RUBY
+
+      actions :create_file, "bin/dev", %{#!/usr/bin/env bash
+
+runner=foreman
+
+if command -v overmind &> /dev/null
+then
+  runner=overmind
+elif ! command -v foreman &> /dev/null
+then
+  echo "Installing foreman..."
+  gem install foreman
+fi
+
+$runner start -f Procfile.dev}
+
+      actions :chmod, "bin/dev", 777
+    end
+
+    private
+
+    class Actions < Thor
```

Comment:
It seems in the other rake tasks we are not using Thor, so is there some specific reason why we need to use it here? The only things that we seem to do are creating files and `chmod`, so I think there's not much need to use Thor here.

But if there is a reason, why not move the actions related code to this class if we are using Thor in the first place?

Reply:
I do like to use Thor in this case as Class is providing the setup, and Thor::Actions provide the prompt to replace, diff, like the one below: 

```
root@4cb16400665a:/code# bin/bundle exec rake decidim:procfile:install
    conflict  Procfile.dev
Overwrite /code/Procfile.dev? (enter "h" for help) [Ynaqdhm] 
```
The chown is just because we are making use of that class, and handles everything related to Rails.root inside the application.